extension/googleclientauthextension/factory.go

```diff
@@ -54,10 +56,26 @@ func (ca *clientAuthenticator) Start(ctx context.Context, _ component.Host) erro
 	if config.QuotaProject == "" {
 		config.QuotaProject = quotaProjectFromCreds(creds)
 	}
-	ca.TokenSource = &oauth.TokenSource{TokenSource: creds.TokenSource}
+	source, err := ca.newTokenSource(ctx, creds)
+	if err != nil {
+		return err
+	}
+	ca.TokenSource = &oauth.TokenSource{TokenSource: source}
 	return nil
 }
 
+func (ca *clientAuthenticator) newTokenSource(ctx context.Context, creds *google.Credentials) (oauth2.TokenSource, error) {
+	switch ca.config.TokenFormat {
+	case "id_token":
+		if ca.config.Audience == "" {
+			return nil, errors.New("audience must be specified")
+		}
+		return idtoken.NewTokenSource(ctx, ca.config.Audience)
+	default:
```

Comment:
we should also update `(cfg *Config) Validate()` to check that the input is one of the valid values and return an error if not

Reply:
I added validation for `token_format`. https://github.com/GoogleCloudPlatform/opentelemetry-operations-go/pull/851/commits/d50d02bdd1648891cf82f8a7cbe86ae0489e54fe